clients/drcachesim/tests/memref_gen.h

```diff
@@ -137,6 +146,34 @@ gen_exit(memref_tid_t tid)
     return memref;
 }
 
+inline std::vector<memref_t>
+get_memrefs_from_ir(instrlist_t *ilist, std::vector<memref_instr_t> &memref_instr_vec,
+                    const addr_t base_addr = BASE_ADDR)
+{
+    static const int MAX_DECODE_SIZE = 1024;
+    byte decode_buf[MAX_DECODE_SIZE];
+    byte *pc =
+        instrlist_encode_to_copy(GLOBAL_DCONTEXT, ilist, decode_buf,
+                                 reinterpret_cast<app_pc>(base_addr), nullptr, true);
+    assert(pc != nullptr);
+    std::vector<memref_t> memrefs = {};
+    memrefs.push_back(
+        gen_marker(1, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS));
+    for (auto pair : memref_instr_vec) {
+        if (pair.instr != nullptr && type_is_instr(pair.memref.instr.type)) {
+            pair.memref.instr.addr = instr_get_offset(pair.instr) + base_addr;
+            pair.memref.instr.size = instr_length(GLOBAL_DCONTEXT, pair.instr);
+            byte buf[MAX_ENCODING_LENGTH];
+            byte *next_pc = instr_encode(GLOBAL_DCONTEXT, pair.instr, buf);
+            assert(next_pc != nullptr);
+            memcpy(pair.memref.instr.encoding, buf, sizeof(buf));
```

Comment:
The size copied looks suspect: should not be some separately defined size; should be either the size of the dest or better the instr length (may want to assert that instr length fits in the .encoding field too)

Reply:
Done.